test/software/amazon/ion/GoodIonTest.java

```diff
@@ -98,6 +99,8 @@ public void test()
             IonReader binaryReader = system().newReader(encoded);
 
             ReaderCompare.compare(treeReader, binaryReader);
+        } else {
+           EquivsTest.roundTripEquivalence(datagram);
```

Comment:
Why can't you just remove the `if (! myFileIsBinary)` check and use the existing logic for both binary and text?